lib/jackrabbit.js

```diff
@@ -94,20 +95,30 @@ const jackrabbit = (url, logger, options = {}) => {
 
     // private
 
-    const bail = (err) => {
+    const bail = (err, newConnectionAttempt = false) => {
 
         // TODO close any connections or channels that remain open
         connection = undefined;
-        if (err && !tryReconnect(err)) {
-            rabbit.emit('error', err);
-            doLog('fatal', 'Rabbit connection error!');
-            process.exit(1);
+        if (!err) {
+            return;
+        }
+
+        if (newConnectionAttempt && tryReconnect(err)) {
+            return;
         }
```

Comment:
@bladesling for this case wouldn't be better to keep it in one if? or the idea is to leave it like this for readibility purposes?
```suggestion
        if (!err || (newConnectionAttempt && tryReconnect(err))) {
            return;
        }
```

Reply:
Yes it is for readability.